### ISSUE TITLE
The Superuser binary (su) must be updated.

### ISSUE THREAD

**Issue by uvsmtid** [koush/Superuser#233]
I installed Superuser = v1.0.3.0 on LG G2 D802:
- CyanogenMod = cm-10.2.0-d802.zip
- Team Win Recovery Project (TWRP)= v2.6.3.3

When I open Superuser, it pops up a window titled "Install" with the following message:

```
The Superuser binary (su) must be updated.

Please choose an installation method.
Recovery mode installation is recommended for HTC devices.
```

The window contains two buttons `Cancel` and `Recovery Install`.
- If `Cancel` is pressed, nothing happens. Reopening Superuser will bring the window back.
- If `Recovery Install` is pressed, the phone _immediately_ reboots into TWRP recovery.

It's absolutely not clear what should be done under recovery mode. Booting up the system from recovery mode and reopening Superuser will bring the window again.

**UPDATE**
This issue seems related to #216 #214 #185 #174.

**Comment by uvsmtid** [comment]
After looking through all hints, I found a workaround to automatic binary update.

###### \- Download application.

On the main app page in [Google Play](https://play.google.com/store/apps/details?id=com.koushikdutta.superuser) there is a link to `zip` file.

```
wget http://download.clockworkmod.com/superuser/superuser.zip
```

###### \- Enable "Developer options"

###### \- Enable "USB debugging"

###### \- Connect device to computer

###### \- Push downloaded `zip` file to device

```
adb push superuser.zip /data/local/tmp/ 
```

###### \- Reboot into recovery

```
adb reboot recovery
```

###### \- Install pushed `zip` file

The popup window does not appear anymore.

**Comment by StormTrooper** [comment]
Was also having the same problem - your workaround seems to have fixed it for me as well. Thanks.

**Comment by Sound-Proof** [comment]
Question, what an I missing when you say boot into recovery? Then reboot
it? Or are you being literal, as in reboot and that recovery will then load?

Please email me at saranoyd@gmail.com for personal emails.

My Profesional email is SaraGroves@me.com

Email me for my new phone number, which has recently changed.

**Comment by bitcoinis** [comment]
Same problem here. I have 1.0.3.0 installed. A few weeks ago it started asking to install the new binary. I have always answered no because it was suspicious that the app was not updated.
I also started having "shell" automatically getting user permission 10-15 times in a row. Why shell does not either have to be authorized  to execute su or can be prevented to get su access?
Is there a way to monitor what command is sent to superuser?
Am I being paranoid or there is something suspicious going on on my phone?

**Comment by ActualHuman1** [comment]
I just rooted my galaxy s4 and when I'm prompted to do a binary update it says that the installation failed what do I do?

**Comment by Sound-Proof** [comment]
I got it to update on my htc, I can't even get su on galaxy

**Comment by x3rAx** [comment]
# Workaround:

See ~> https://github.com/koush/Superuser/issues/216#issuecomment-38354585

---

This seems to be related the following: #216, #214, #185, #174

**Comment by Bryanshack** [comment]
Ok can anybody help when I first got my s4 1337 I rooted it .and then I update it not thanking to 4.42 . I have a superuser app installed on my phone which I can no delete .  It says no Su binary installed and super su cannot install it .this is a problem. If you just upgraded to android 4.3 you need to manually re-root consult the relevant forums of your device . Ive tryed re-rooting but nothing seems to work

**Comment by kvp2242** [comment]
Quick question- when I download the zip file, where should I place it? I know it probably shouldn't stay in my downloads folder.

**Comment by Hasanall** [comment]
Hello
I root my device by e root & I use it successfully last 3 month. But now phone is not rooted & I face this problem. What can I solve this problem. Pls help. My whatsapp +8801934080798

**Comment by Hasanall** [comment]
Pls give me your facebook id . It will be helpful for us

**Comment by Peter-Pan-7** [comment]
Hi,

thanks to uvsmtid for the tip: it did help me fixing this issue with superuser from koush.

I can say than one year later this issue remains with this irrelevant message always showing up after boot and asking for an update.

My concern was that my XperIia arc S cannot boot in recovery mode using the reboot to recovery option of CWM from pvyParts or using an adb reboot recovery command.

I suppose that CWM did not install properly. So I was stuck with no means to fix this superuser issue using adb commands. Based on tips provided by uvsmtid, I did fix this issue this way as I am a Windows user.

wget http://download.clockworkmod.com/superuser/superuser.zip => download the ZIP file using firefox
- Enable "Developer options"
- Enable "USB debugging"
- Connect device to computer
- Push downloaded zip file to device

adb push superuser.zip /data/local/tmp/  => upload the file ZIP to the phone using a bluetooth connection (stored by default in folder /mnt/sdcard/bluetooth) then move the file to /data/local/tmp/ using ES File Explorer (this second step in not mandatory)
- Reboot into recovery

adb reboot recovery => reboot in recovery mode manually while pressing repeatedly any phone button during early steps of boot (logo "Sony Ericssson" showing up)
- Install pushed zip file => the same: "install zip from sdcard" then "choose zip from sdcard" then select the superuser.zip file while navigating through the folders (in  /mnt/sdcard/bluetooth for me)

The popup window does not appear anymore => yeah! pretty cool!

Thanks again!

**Comment by ghost** [comment]
When I follow these instructions, it fails when trying to verify the update package with the following messages:

> E: footer is wrong
> E: signature verification failed

My phone is a Verizon Galaxy S4, build VRUFNK1, rooted with [these instructions](http://forum.xda-developers.com/galaxy-s4-verizon/general/how-to-root-i545vrufnk1-4-4-2-12-1-2014-t2959613). I then uninstalled the SuperSU app (from within the SuperSU app, "install other SU manager app" function) and installed this app.

Can anyone help?

---

Edit: it seems others have had success by manually updating the binary, presumably by unpacking the zip file and copying its contents to the right locations using their file manager that has root acess. I will update this post when I dermine which files must be copied where.

**Comment by mki2** [comment]
For me, re-mounting /system for rw helped:
> adb shell
$ su
# mount | grep system
Note the device path for mounting
# mount -o rw,remount $devicepath /system
Then su can be updated manually from the installed apk path.

**Comment by aniket8855** [comment]
I just rooted my Nexus 5 ..and installed cm13 with Gapps nano 
Then I was it in a video how to install superuser..it said download d latest version of superuser from its website..and flash d zip file
While flashing d zip file..it was successful ..but I got this big hilighted msg saying if the su ask for binary update _DO NOT DO IT, YOU WILL BREAK THINGS_ so should I update binary or no? 
Please help!!!

**Comment by Bryanshack** [comment]
Yes update  binary  ,as normal let it do its thing and if it freezes  just
start over

**Comment by aniket8855** [comment]
Thanks man (y) it worked..thanks a lot

**Comment by gripedthumbtacks** [comment]
Downloading an su binary over unencrypted HTTP is a horrible idea and your phone will get owned. At least most to HTTPS, add some checksums like SHA-256, or digitally sign the binary.

**Comment by gmahadhika** [comment]
Hi, I've followed all instructions by uvsmtid, but Superuser still asks to update binaries. My device is Terminated Note 4 (MTK). Please help.

**Comment by Bryanshack** [comment]
Update binary on normal mode after a little while it should reboot,  and
everything  should be ok
